Remove all uses of PUBLIC_IN_WEBGPU_SWIFT

geoffreygaren opened the pull request:
#### 1d2530a18d24e4f56c87e4bcd3f3d716e944cca6
<pre>
Remove all uses of PUBLIC_IN_WEBGPU_SWIFT
<a href="https://bugs.webkit.org/show_bug.cgi?id=299042">https://bugs.webkit.org/show_bug.cgi?id=299042</a>
<a href="https://rdar.apple.com/160803645">rdar://160803645</a>

Reviewed by Mike Wyrzykowski and Elliott Williams.

PUBLIC_IN_WEBGPU_SWIFT was a glorious hack to enable implementing parts of a C++
class in Swift despite name lookup rules making private and protected members
invisible.

Now that <a href="https://rdar.apple.com/137764620">rdar://137764620</a> is fixed, we can use SWIFT_PRIVATE_FILEID instead.

I consider this a subtask of removing un-safety in this code because making
private fields public is... not a great way to uphold essential invariants.

* Configurations/CommonBase.xcconfig: Enable SWIFT_PRIVATE_FILEID
* Source/WebGPU/WebGPU/Buffer.h: Use SWIFT_PRIVATE_FILEID instead of public.
* Source/WebGPU/WebGPU/CommandEncoder.h: Use SWIFT_PRIVATE_FILEID instead of public.
* Source/WebGPU/WebGPU/CommandsMixin.h: Declare our enum public because it
doesn&apos;t hold any private state, and all subclasses inherit this class publicly
anyway. (Also, we can&apos;t support protected enum declarations until
<a href="https://rdar.apple.com/160803362">rdar://160803362</a> is fixed.)
* Source/WebGPU/WebGPU/Device.h: No need for double `private` anymore since
there&apos;s no interstitial `public`.
(WebGPU::Device::nopVertexFunction const):
* Source/WebGPU/WebGPU/Queue.h: Use SWIFT_PRIVATE_FILEID instead of public.
* Source/WebGPU/WebGPU/WebGPUExt.h: No need for SWIFT_PRIVATE_FILEID anymore!

Canonical link: <a href="https://commits.webkit.org/300128@main">https://commits.webkit.org/300128@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/34ea499922424e68b607dd42f760aa394df17996

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/121464 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/41161 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/31820 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/127913 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/73547 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/5bebc2b6-8f2c-4110-a23b-6115211d5679) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/41863 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/49740 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/92253 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/73547 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/36abafc9-5772-44b8-81db-de41b6e6e6b0) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/124416 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/33424 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/108810 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/72929 "Passed tests") | | ⏳ 🛠 vision-apple 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/32433 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/71485 "Built successfully") | | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/102912 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/27146 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/130739 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/48392 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/36790 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/100843 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/48760 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/105031 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/100750 "Passed tests") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/46162 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/24242 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/45088 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/19253 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/48251 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/53963 "Built successfully") | | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/125/builds/47722 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/129/builds/51068 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/124/builds/49404 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
<!--EWS-Status-Bubble-End-->

emw-apple reviewed (approved):
rs=me, good riddance!

geoffreygaren committed:
Remove all uses of PUBLIC_IN_WEBGPU_SWIFT
https://bugs.webkit.org/show_bug.cgi?id=299042
rdar://160803645

Reviewed by Mike Wyrzykowski and Elliott Williams.

PUBLIC_IN_WEBGPU_SWIFT was a glorious hack to enable implementing parts of a C++
class in Swift despite name lookup rules making private and protected members
invisible.

Now that rdar://137764620 is fixed, we can use SWIFT_PRIVATE_FILEID instead.

I consider this a subtask of removing un-safety in this code because making
private fields public is... not a great way to uphold essential invariants.

* Configurations/CommonBase.xcconfig: Enable SWIFT_PRIVATE_FILEID
* Source/WebGPU/WebGPU/Buffer.h: Use SWIFT_PRIVATE_FILEID instead of public.
* Source/WebGPU/WebGPU/CommandEncoder.h: Use SWIFT_PRIVATE_FILEID instead of public.
* Source/WebGPU/WebGPU/CommandsMixin.h: Declare our enum public because it
doesn't hold any private state, and all subclasses inherit this class publicly
anyway. (Also, we can't support protected enum declarations until
rdar://160803362 is fixed.)
* Source/WebGPU/WebGPU/Device.h: No need for double `private` anymore since
there's no interstitial `public`.
(WebGPU::Device::nopVertexFunction const):
* Source/WebGPU/WebGPU/Queue.h: Use SWIFT_PRIVATE_FILEID instead of public.
* Source/WebGPU/WebGPU/WebGPUExt.h: No need for SWIFT_PRIVATE_FILEID anymore!

Canonical link: https://commits.webkit.org/300128@main

adetaylor reviewed (commented):
LGTM
